### PR TITLE
Clean up and Document Placement 

### PR DESCRIPTION
**Pull request by saaramahmoudi**
#### Description
Small PR; placer seed was always printed as 0 in vpr.out, since It was printed before It was set. Hence, the default value regardless of the command-line was printed. Set it before printing. 

#### Types of changes

- [x] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Checklist:

- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

**Comment by saaramahmoudi** [comment]
@vaughnbetz

**Comment by saaramahmoudi** [comment]
@vaughnbetz can you please publish your review so I can take actions?

File: vpr/src/base/SetupVPR.cpp [hunk 1]
```diff
@@ -666,6 +666,8 @@ static void SetupPlacerOpts(const t_options& Options, t_placer_opts* PlacerOpts)
     PlacerOpts->place_constraint_subtile = Options.place_constraint_subtile;
     PlacerOpts->floorplan_num_horizontal_partitions = Options.floorplan_num_horizontal_partitions;
     PlacerOpts->floorplan_num_vertical_partitions = Options.floorplan_num_vertical_partitions;
+
+    PlacerOpts->seed = Options.Seed;
```
[the remark this answers is not in the document]
Comment (vaughnbetz) [reply vpr/src/base/SetupVPR.cpp:669]: @saaramahmoudi : placerOpts->seed setting line is now removed from the placer. If the two documentation updates can be made we can merge this. Or if that will take a while please create an issue to track them and I'll merge this.

**Comment by saaramahmoudi** [comment]
@vaughnbetz The RLPlace command-line options are now added to the documentation. I also added NoC options to documentation since they weren't available either and fixed some traffic flow typos.

File: doc/src/vpr/command_line_usage.rst [hunk 2]
```diff
@@ -920,6 +980,82 @@ The following options are only valid when the placement engine is in timing-driv
 
     Name of the post-placement timing report file to generate (not generated if unspecfied).
 
+
```
Comment (vaughnbetz) [inline doc/src/vpr/command_line_usage.rst:983]: I know @Srivat97 has a PR that's about to land that documents the NoC options too. @Srivat97 : I suggest landing yours and then Sara can enhance any NoC documentation that doesn't capture everything she has here.

File: doc/src/vpr/command_line_usage.rst [hunk 3]
```diff
@@ -821,6 +821,66 @@ Setting any of the following options selects `Dusty's annealing schedule <dusty_
 
     **Default:** ``0.25``
 
+.. option:: --place_cost_exp <float>
+
+    Wiring cost is divided by the average channel width over a net's bounding box
+    taken to this exponent.Only impacts devices with different channel widths in
```
Comment (vaughnbetz) [inline doc/src/vpr/command_line_usage.rst:827]: Add a space after "."

File: doc/src/vpr/command_line_usage.rst [hunk 4]
```diff
@@ -821,6 +821,66 @@ Setting any of the following options selects `Dusty's annealing schedule <dusty_
...
+    Wiring cost is divided by the average channel width over a net's bounding box
+    taken to this exponent.Only impacts devices with different channel widths in 
+    different directions or regions. 
+
+    **Default:** ``1``
+
+.. option:: --RL_agent_placement {on | off}
+
+    Uses a Reinforcement Learning (RL) agent in choosing the appropiate move type in placement.
+    It activates the RL agent placement instead of using fixed probability for each move type.
```
Comment (vaughnbetz) [inline doc/src/vpr/command_line_usage.rst:835]: "a" fixed probability for each

File: doc/src/vpr/command_line_usage.rst [hunk 5]
```diff
@@ -821,6 +821,66 @@ Setting any of the following options selects `Dusty's annealing schedule <dusty_
...
+
+    **Default:** ``on``
+
+.. option:: --place_agent_multistate {on | off}
+
+    Enable multistate agent in the placement. A second state will be activated late in
```
Comment (vaughnbetz) [inline doc/src/vpr/command_line_usage.rst:841]: "Enable a multistate agent during RL placement."  (first sentence).

File: doc/src/vpr/command_line_usage.rst [hunk 6]
```diff
@@ -821,6 +821,66 @@ Setting any of the following options selects `Dusty's annealing schedule <dusty_
...
+
+.. option:: --place_agent_algorithm {e_greedy | softmax}
+
+    Controls which placement RL agent is used. 
+
+    **Default:** ``softmax``
+
+.. option:: --place_agent_epsilon <float>
+
+    Placement RL agent's epsilon for epsilon-greedy agent. Epsilon represents
```
Comment (vaughnbetz) [inline doc/src/vpr/command_line_usage.rst:854]: "for the epsilon-greedy agent."

File: doc/src/vpr/command_line_usage.rst [hunk 7]
```diff
@@ -821,6 +821,66 @@ Setting any of the following options selects `Dusty's annealing schedule <dusty_
...
+    Controls how quickly the agent's memory decays. Values between [0., 1.] specify
+    the fraction of weight in the exponentially weighted reward average applied to moves
+    which occured greater than moves_per_temp moves ago. Values < 0 cause the
+    unweighted reward sample average to be used (all samples are weighted equally)
+
+    **Default:** ``0.05``
+
+.. option:: --place_reward_fun {basic | nonPenalizing_basic | runtime_aware | WLbiased_runtime_aware}
+
+    The reward function used by placement RL agent to learn best action at each anneal stage.
```
Comment (vaughnbetz) [inline doc/src/vpr/command_line_usage.rst:870]: "by the placement RL agent to learn the best action"  (missing a couple of "the" )

File: doc/src/vpr/command_line_usage.rst [hunk 8]
```diff
@@ -821,6 +821,66 @@ Setting any of the following options selects `Dusty's annealing schedule <dusty_
...
+
+    The reward function used by placement RL agent to learn best action at each anneal stage. 
+
+    .. note:: The latter two are only available for timing-driven placement. 
+    
+    **Default:** ``WLbiased_runtime_aware``
+
+.. option:: --place_agent_space {move_type | move_block_type}
+
+    Agent exploration space can be either based on only move types or also consider different block types.
```
Comment (vaughnbetz) [inline doc/src/vpr/command_line_usage.rst:878]: "The RL agent exploration space can either be based only on move types or can also consider the block type being moved."

**Review by vaughnbetz — CHANGES_REQUESTED** [review]
Looks good, except for a few wording nits.  
Also, @Srivat97 has simultaneously been making a PR for NoC documentation so you'll need to sync up with him on that part.

**Comment by saaramahmoudi** [comment]
> Looks good, except for a few wording nits. Also, @Srivat97 has simultaneously been making a PR for NoC documentation so you'll need to sync up with him on that part.

I don't see a PR from @Srivat97. Can you put the PR's name here and I can merge the branches and land both together?

**Comment by vaughnbetz** [comment]
@Srivat97 : probably you haven't made a PR yet (@saaramahmoudi : Srivatsan showed me the documentation in our meeting today, but I don't see the PR either). Can you make a PR @Srivat97 ?

**Comment by saaramahmoudi** [comment]
@vaughnbetz I just did a small clean-up on NoC router description in vpr.out, I moved printing of router connections in the architecture to echo file instead of printing 100 lines in the vpr.out. Could you please take a look at it to see if that sounds good to you?

**Comment by vaughnbetz** [comment]
Looks good, thanks.  Will merge after talking to Srivatsan today.

**Comment by saaramahmoudi** [comment]
@vaughnbetz ready to be merged.

**Comment by Srivat97** [comment]
@vaughnbetz , @saaramahmoudi : Sorry for the delayed response. You can go ahead and merge this PR first. I'll handle the conflicts on my end.

**Pull request merged**